kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/ChecksumSimpleComparisonSuite.scala

```diff
@@ -0,0 +1,338 @@
+        .commit(engine, emptyIterable())
+        .getPostCommitHooks
+        .forEach(hook => hook.threadSafeInvoke(engine))
+      spark.sql(s"CREATE OR REPLACE TABLE delta.`${sparkTablePath}` (id Integer) USING DELTA")
+      assertChecksumEquals(engine, sparkTablePath, kernelTablePath, 0)
+
+      (1 to 10).foreach(
+        version => insertIntoUnpartitionedTableAndCheckCrc(
+          engine, sparkTablePath, kernelTablePath, version)
+      )
+
+    }
+  }
+
+  test("create partitioned table, insert and verify checksum") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val sparkTablePath = tablePath + "spark"
+      val kernelTablePath = tablePath + "kernel"
+      Table
+        .forPath(engine, kernelTablePath)
+        .createTransactionBuilder(engine, "test-engine", CREATE_TABLE)
+        .withSchema(engine, new StructType().add("id", INTEGER).add(PARTITION_COLUMN, INTEGER))
+        .withPartitionColumns(engine, Seq(PARTITION_COLUMN).asJava)
+        .build(engine)
+        .commit(engine, emptyIterable())
+        .getPostCommitHooks
+        .forEach(hook => hook.threadSafeInvoke(engine))
+      spark.sql(
+        s"CREATE OR REPLACE TABLE delta.`${sparkTablePath}` " +
+        s"(id Integer, part Integer) USING DELTA PARTITIONED BY (part)"
+      )
+      assertChecksumEquals(engine, sparkTablePath, kernelTablePath, 0)
+
+      (1 to 10).foreach(
+        version => insertIntoPartitionedTableAndCheckCrc(
+          engine, sparkTablePath, kernelTablePath, version)
+      )
+    }
+  }
+
+  private def assertChecksumEquals(
+      engine: Engine,
+      sparkTablePath: String,
+      kernelTablePath: String,
+      version: Long): Unit = {
+    val sparkCrcPath = buildCrcPath(sparkTablePath, version)
+    val kernelCrcPath = buildCrcPath(kernelTablePath, version)
+
+    assert(
+      Files.exists(sparkCrcPath) && Files.exists(kernelCrcPath),
+      s"CRC files not found for version $version"
+    )
+
+    val sparkCrc = readCrcInfo(engine, sparkTablePath, version)
+    val kernelCrc = readCrcInfo(engine, kernelTablePath, version)
+
+    assertCrcInfoEquals(sparkCrc, kernelCrc)
+  }
+
+  private def readCrcInfo(engine: Engine, path: String, version: Long): CRCInfo = {
+    ChecksumReader
+      .getCRCInfo(engine, new Path(s"$path/_delta_log"), version, version)
+      .orElseThrow(() => new IllegalStateException(s"CRC info not found for version $version"))
+  }
+
+  private def buildCrcPath(basePath: String, version: Long): java.nio.file.Path = {
+    new File(f"$basePath/_delta_log/$version%020d.crc").toPath
+  }
+
+  private def assertCrcInfoEquals(crc1: CRCInfo, crc2: CRCInfo): Unit = {
+    assert(crc1.getVersion === crc2.getVersion)
+    assert(crc1.getNumFiles === crc2.getNumFiles)
+    assert(crc1.getTableSizeBytes === crc2.getTableSizeBytes)
+    assert(crc1.getMetadata.getSchema === crc2.getMetadata.getSchema)
+    assert(crc1.getMetadata.getPartitionColNames === crc2.getMetadata.getPartitionColNames)
+  }
```

Comment:
can you just add equals/hashCode to CRCInfo?

Reply:
Metadata and Protocol doesn't have equals/hashCode implemented so adding equals/hashCode to CRCInfo won't works without fixing Metadata and Protocol.

Added a todo and I will make it in a separate PR